readthedocs/api/v3/serializers.py

```diff
@@ -778,18 +778,23 @@ class Meta:
 
         expandable_fields = {
             # NOTE: this has to be a Model method, can't be a
-            # ``SerializerMethodField`` as far as I know
+            # ``SerializerMethodField`` as far as I know.
+            # NOTE: this lists public versions only.
             "active_versions": (
                 VersionSerializer,
                 {
                     "many": True,
                 },
             ),
+            # NOTE: we use a serializer without expandable fields to avoid
+            # leaking information about the organization through the project.
             "organization": (
-                "readthedocs.api.v3.serializers.OrganizationSerializer",
+                "readthedocs.api.v3.serializers.OrganizationSerializerWithoutExpandableFields",
                 # NOTE: we cannot have a Project with multiple organizations.
                 {"source": "organizations.first"},
             ),
+            # NOTE: we are leaking the slugs of all teams linked to this project
+            # to anyone with access to this prtoject. It's only the slug, but still.
             "teams": (
```

Comment:
I think we can just have a `teams` method field (expanded by default, but with all the correct permissions checked) or a `projects/{project}/teams/` endpoint.

Reply:
I understand it isn't possible to have permission check on expandable fields, right?

Reply:
yeah, it's hard/impossible to filter related fields of an object using expandable fields. This is since the related field doesn't know about the current user or whatever checks we need, and we can't override that.

We could implement our own expandable fields, but in a more limited way, like just for detail endpoints (implementing the checks on list views may have an impact on performance), and without recursion (just allow expanding one level).

This also reminds me of this post about graphql https://bessey.dev/blog/2024/05/24/why-im-over-graphql/